### PR TITLE
[15.0] [IMP] sale_advance_payment

### PR DESCRIPTION
**Pull request by Nikul-OSI**

### CODE REVIEW

File: sale_advance_payment/models/account_move.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
```
Comment (dreispt) [inline sale_advance_payment/models/account_move.py:2]: Add copyright line

Comment (dreispt) [reply sale_advance_payment/models/account_move.py:2]: Also add to the Contributors file

**Comment by dreispt** [comment]
@Nikul-OSI Code looks good, but tests are failing. Please also add a test for this new code.

File: sale_advance_payment/tests/test_sale_advance_payment.py [hunk 2]
```diff
@@ -361,32 +361,33 @@ def test_03_residual_amount_big_pre_payment(self):
...
+        # ) = pre_payment._seek_for_lines()
+        # (
+        #     counterpart_lines
+        #     + invoice.line_ids.filtered(
+        #         lambda line: line.account_internal_type == "receivable"
+        #     )
+        # ).reconcile()
+        # self.sale_order_1.invalidate_cache()
+        # self.assertEqual(self.sale_order_1.amount_residual, 1300)
+        # self.assertEqual(invoice.amount_residual, 0)
```
Comment (dreispt) [inline sale_advance_payment/tests/test_sale_advance_payment.py]: The reconciliation is automatic, so it makes sense to delete that code.
But the check for the final open amount should stay, so keep these two lines please.

File: sale_advance_payment/tests/test_sale_advance_payment.py [hunk 3]
```diff
@@ -294,21 +294,21 @@ def test_02_residual_amount_with_invoice(self):
...
+        #     writeoff_lines,
+        # ) = pre_payment._seek_for_lines()
+        # (
+        #     counterpart_lines
+        #     + invoice.line_ids.filtered(
+        #         lambda line: line.account_internal_type == "receivable"
+        #     )
+        # ).reconcile()
+        # self.sale_order_1.invalidate_cache()
+        # self.assertEqual(self.sale_order_1.amount_residual, 2200)
```
Comment (dreispt) [inline sale_advance_payment/tests/test_sale_advance_payment.py]: You can delete the lines, this part of the test is not needed anymore.

**Comment by dreispt** [comment]
@omar7r We are proposing a functional change her, and would like your PoV on it.
The rational is, when entering a pre-Payment for the Sales Order, the users expect it to be automatically applied to the corresponding invoice.
Thanks!

**Comment by rousseldenis** [comment]
@Nikul-OSI Thanks for this. Could you add a description of your enhancement ?

File: sale_advance_payment/models/account_move.py [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+import json
+
+from odoo import models
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    def action_post(self):
+        res = super(AccountMove, self).action_post()
```
Comment (rousseldenis) [inline sale_advance_payment/models/account_move.py:14]: Could you add some comments ? Thanks

Comment (dreispt) [reply sale_advance_payment/models/account_move.py:14]: Comments? Can you elaborate?

**Review by dreispt — COMMENTED** [review]
Tested and looks beautiful!

**Review by dreispt — APPROVED** [review]

**Review by omar7r — APPROVED** [review]
@dreispt It's fine, good improvement. Tested.

**Pull request merged**